### ISSUE TITLE
UseSelection 

### ISSUE THREAD

**Issue by PascalSenn** [ChilliCream/hotchocolate#1164]
We could use the power of graphql to only select the data we actually need fom the database. 

```csharp
descriptor.Field(t => t.GetSomething())
   .UsePaging<FooType>()
   .UseSelection()
   .UseFiltering()
   .UseSorting();
````

**Comment by michaelstaib** [comment]
With paging we this is a little more difficult since we have to aggregate the actual fields selected on `nodes` and `edges`. Just as a reminder.

**Comment by PascalSenn** [comment]
Done as it is implemented in 10.4 and 11
